tests/test_other.py

```diff
@@ -10584,3 +10584,8 @@ def test_concepts(self):
   def test_link_only_setting_warning(self):
     err = self.run_process([EMCC, '-sALLOW_MEMORY_GROWTH', '-c', test_file('hello_world.c')], stderr=PIPE).stderr
     self.assertContained("warning: linker setting ignored during compilation: 'ALLOW_MEMORY_GROWTH' [-Wunused-command-line-argument]", err)
+
+  def test_explict_gl_linking(self):
+    # ensure libgl exists
+    self.run_process([EMBUILDER, 'build', 'libgl'])
```

Comment:
No.  As of today if you use  `NO_AUTO_NATIVE_LIBRARIES` there is no way to trigger the auto-building of the affected libraries (libgl, libal, libhtml5).   I think we can fix that as part of #14341

Reply:
(this PR doesn't change that)

Reply:
`NO_AUTO_NATIVE_LIBRARIES` was always there.

And yes I can not remove the line above your are right.  Sorry about that. 

The reason that `NO_AUTO_NATIVE_LIBRARIES` is needed is because `-lGL` was previously failing in that mode (`NO_AUTO_NATIVE_LIBRARIES` is enabled in strict mode which is more common than the almost unknown/unused `NO_AUTO_NATIVE_LIBRARIES`).

Before:

`-sSTRING -lGL` -> link failure because to interpret `-lGL` as being just about JS libraries

After:

`-sSTRICT -lGL` -> link success because we find and use the native library.